lib/backend/etcdbk/etcd.go

```diff
@@ -217,9 +218,25 @@ func New(ctx context.Context, params backend.Params) (*EtcdBackend, error) {
 		watchDone:        make(chan struct{}),
 		buf:              buf,
 	}
+
 	if err = b.reconnect(); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	// Check that the etcd nodes are at least the minimum version supported
+	timeout, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
+	for _, n := range cfg.Nodes {
+		status, err := b.client.Status(timeout, n)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		if status.Version < teleport.MinimumEtcdVersion {
```

Comment:
I'm not sure if this is a reliable way to compare the versions as it doesn't know anything about what a semver is - it's just doing a string comparison.

Have a look at the way `clientv3` does it - they're extracting the major/minor version parts and doing a complete comparison: https://github.com/gravitational/teleport/blob/master/vendor/go.etcd.io/etcd/clientv3/client.go#L521-L570

We also have version comparison logic for our own clients here using `github.com/coreos/go-semver/semver`: https://github.com/gravitational/teleport/blob/master/lib/utils/ver.go#L27-L51

Reply:
Yeah you're absolutely right, my comparison only works for single digit number version parts. I'll find a way to properly compare semver #'s.

Reply:
It looks like the changes between 3.2 and 3.3 were reasonably substantial: https://etcd.io/docs/v3.3.12/upgrades/upgrade_3_3/

I'd probably stick with us requiring 3.3+.